src/main/scala/viper/carbon/modules/impls/DefaultStateModule.scala

```diff
@@ -149,6 +150,14 @@ class DefaultStateModule(val verifier: Verifier) extends StateModule {
   }
 
   override def freshTempStateKeepCurrent(name: String) : StateSnapshot = {
+    freshTempStateKeepCurrentAux(name, false)
+  }
+
+  private def freshTempStateKeepCurrentAux(name: String, usedForOldState: Boolean) : StateSnapshot = {
+    if(name.equals("old") && !usedForOldState) {
+      sys.error("freshTempStateKeepCurrent invoked with reserved \"old\" name")
```

Comment:
I was going to suggest using ``assert(...)`` instead of ``sys.error`` for things like this (I assume this is never supposed to fail), but apparently Carbon mostly uses ``sys.error`` except for where it doesn't, so I guess this is fine.

Reply:
Good point. We could change the `sys.error` to `assert` convention change in another PR.

Reply:
I changed the code to use `assert` for the instances of this PR.